modules/rostests/kmtests/ntos_ob/ObInsert.c

```diff
@@ -0,0 +1,651 @@
+/*
+ * PROJECT:         ReactOS kernel-mode tests
+ * LICENSE:         GPLv2+ - See COPYING in the top level directory
+ * PURPOSE:         Kernel-Mode Test Suite - ObInsertObject test
+ * PROGRAMMER:      Gleb Surikov <glebs.surikovs@gmail.com>
+ */
+
+#include <kmt_test.h>
+
+#define NDEBUG
+#include <debug.h>
+
+typedef struct DUMMY_TYPE
+{
+    ULONG Data;
+} DUMMY_TYPE, *PDUMMY_TYPE;
+
+static HANDLE DirectoryHandle;
+static UNICODE_STRING DirectoryName;
+
+static POBJECT_TYPE DummyType;
+static UNICODE_STRING DummyTypeName;
+static OBJECT_TYPE_INITIALIZER ObjectTypeInitializer;
+static OBJECT_ATTRIBUTES DummyObjectAttributes;
+
+static
+VOID
+NTAPI
+DumpProc(
+    IN PVOID Object,
+    IN POB_DUMP_CONTROL DumpControl
+)
+{
+    DPRINT("DumpProc() called\n");
+}
+
+static
+NTSTATUS
+NTAPI
+OpenProc(
+    IN OB_OPEN_REASON OpenReason,
+    IN PEPROCESS Process,
+    IN PVOID Object,
+    IN ACCESS_MASK GrantedAccess,
+    IN ULONG HandleCount
+)
+{
+    DPRINT("OpenProc() 0x%p, OpenReason %d, HandleCount %lu, AccessMask 0x%lX\n",
```

Comment:
I don't know whether you plan to instrumentalize these different callback procedures (Open, Close, etc.) in order to test that they are correctly called whenever an object of your custom type gets operated upon.

Reply:
Removed optional callback procedures in https://github.com/reactos/reactos/pull/6968/commits/38de371bdf839b8cdbefe958e4a61ef3a0c674e9